adam-core/src/main/scala/org/bdgenomics/adam/rich/RichGenotype.scala

```diff
@@ -26,16 +26,56 @@ object RichGenotype {
 }
 
 class RichGenotype(val genotype: Genotype) {
+
   def ploidy: Int = genotype.getAlleles.size
 
   def getType: GenotypeType = {
-    assert(ploidy <= 2, "getType only meaningful for genotypes with ploidy <= 2")
-    genotype.getAlleles.toList.distinct match {
-      case List(GenotypeAllele.Ref) => GenotypeType.HOM_REF
-      case List(GenotypeAllele.Alt) => GenotypeType.HOM_ALT
-      case List(GenotypeAllele.Ref, GenotypeAllele.Alt) |
-        List(GenotypeAllele.Alt, GenotypeAllele.Ref) => GenotypeType.HET
-      case _ => GenotypeType.NO_CALL
+
+    // Get the list with the distinct alleles
+    val distinctListOfAlleles = genotype.getAlleles.toList.distinct
+
+    // In the case that there is only one distinct allele
+    // This should be applicable to any genome ploidy.
+    if (distinctListOfAlleles.size == 1) {
+      distinctListOfAlleles match {
+
+        // If all alleles are the reference allele, the genotype is Homozygous Reference (HOM_REF)
+        case List(GenotypeAllele.Ref)      => GenotypeType.HOM_REF
+
+        // If all alleles are the primary alternative allele, the genotype is Homozygous Alternative (HOM_ALT)
+        case List(GenotypeAllele.Alt)      => GenotypeType.HOM_ALT
+
+        // If all alleles are not called, the genotype is not called  (NO_CALL)
+        case List(GenotypeAllele.NoCall)   => GenotypeType.NO_CALL
+
+        // If all alleles are OtherAlt.
+        // If genotype.getAlleles returns a single OtherAlt, the genotype is Homozygous Alternative (HOM_ALT)
+        // If genotype.getAlleles returns a multiple OtherAlt, the genotype is
+        // A) The OtherAlt alleles are the same OtherAlt alleles: Homozygous Alternative (HOM_ALT)
+        // B) The OtherAlt allales are different OtherAlt alleles: Heterozygous
+        // For now return NO_CALL as the genotypes, as was done in the previous getType function
+        // See also issue https://github.com/bigdatagenomics/adam/issues/897
+        case List(GenotypeAllele.OtherAlt) => GenotypeType.NO_CALL
+
+        // only the four above alleles are possible
+        // https://github.com/bigdatagenomics/bdg-formats/blob/master/src/main/resources/avro/bdg.avdl#L464
+        case _                             => throw new Exception("This should be impossible. Found single distinct allele other than the four possible alleles: Ref, Alt, NoCall and OtherAlt  ")
```

Comment:
Yes thank you for the advice. I'll update this exception in a new commit accordingly.